src/easylink/step.py

```diff
@@ -150,6 +144,32 @@ def validate_step(self, step_config: LayeredConfigTree) -> Dict[str, List[str]]:
             ]
         return errors
 
+    def get_implementation_node_name(self, step_config):
+        """Resolve a sensible unique node name for the implementation graph.
+        This method compares the step node names with the step names through the step hierarchy and
+        uses the full suffix of step names starting from wherever the two first differ. For example,
+        if we have node names step_3_loop_1 step_3_python_pandas the resulting implementation node name
```

Comment:
I think an example in the docstring of where there are duplicate names would be useful

Reply:
done